### PR TITLE
Cumulative set of fixes for os-brick and NFS connector

### PR DESCRIPTION
**Pull request by deiter**
**Description**: Cumulative set of fixes for os-brick NFS connector

* 147390 After deleting the cinder volume still it shows as mounted on controller
* ES-497 After deleting the cinder volume still it shows as mounted on controller
* NEX-23056 After deleting the cinder volume still it shows as mounted on controller
* NEX-23057 Race condition on mount/unmount Cinder NFS volumes
* NEX-23058 NFS 4.1 mont attempts fails

Signed-off-by: Alex Deiter <alex.deiter@gmail.com>

**References**

* Cinder BUG: https://bugs.launchpad.net/cinder/+bug/1559342
* OS-Brick BUG: https://bugs.launchpad.net/os-brick/+bug/1543528

**Pep8 test results**
```
$ tox -e pep8
pep8 create: /opt/stack/os-brick/.tox/pep8
pep8 installdeps: -c/opt/stack/os-brick/../requirements/upper-constraints.txt, -r/opt/stack/os-brick/requirements.txt, -r/opt/stack/os-brick/test-requirements.txt
pep8 develop-inst: /opt/stack/os-brick
pep8 run-test-pre: PYTHONHASHSEED='441112211'
pep8 run-test: commands[0] | flake8
___________________________________ summary ____________________________________
  pep8: commands succeeded
  congratulations :)
```

**Tempest test for NexentaStor5 NFS backend with Python 2.7**
```
$ tox -e ci -- volume
======
Totals
======
Ran: 253 tests in 9354.0000 sec.
 - Passed: 247
 - Skipped: 6
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 7020.9298 sec.
________________________________________________________________________________ summary ________________________________________________________________________________
  ci: commands succeeded
  congratulations :)

```

**Tempest test for NexentaStor5 NFS backend with Python 3.5**
```
$ tox -e ci -- volume
======
Totals
======
Ran: 253 tests in 9592.7604 sec.
 - Passed: 247
 - Skipped: 6
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 7202.3199 sec.
________________________________________________________________________________ summary ________________________________________________________________________________
  ci: commands succeeded
  congratulations :)
```

**Tempest test for NexentaStor4 NFS backend with Python 2.7**
```
$ tox -e ci -- volume
======
Totals
======
Ran: 250 tests in 8932.0000 sec.
 - Passed: 241
 - Skipped: 9
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 6619.7647 sec.
________________________________________________________________________________ summary ________________________________________________________________________________
  ci: commands succeeded
  congratulations :)

```

**Tempest test for NexentaStor4 NFS backend with Python 3.5**
```
$ tox -e ci -- volume
======
Totals
======
Ran: 250 tests in 8992.7097 sec.
 - Passed: 241
 - Skipped: 9
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 6631.7591 sec.
________________________________________________________________________________ summary ________________________________________________________________________________
  ci: commands succeeded
  congratulations :)
```

### CODE REVIEW

**Review by EvgeniiIvanov — APPROVED** [review]
Looks good!

File: os_brick/initiator/connectors/remotefs.py [hunk 1]
```diff
@@ -105,16 +115,24 @@ def connect_volume(self, connection_properties):
         return {'path': path}
 
     @utils.trace
+    @synchronized('disconnect_volume')
     def disconnect_volume(self, connection_properties, device_info,
                           force=False, ignore_errors=False):
-        """No need to do anything to disconnect a volume in a filesystem.
+        """Ensure that the filesystem containing the volume is unmounted.
```
Comment (Qeas) [inline os_brick/initiator/connectors/remotefs.py]: Maybe add a note here that this is only for those volumes that are in separate mounted filesystems?

**Review by Qeas — APPROVED** [review]
Just a minor inline suggestion. Approved.

File: os_brick/initiator/connectors/remotefs.py (hunk 1 above)
Comment (deiter) [reply os_brick/initiator/connectors/remotefs.py]: Sure, I'll update the comment, thank you!

Comment (deiter) [reply os_brick/initiator/connectors/remotefs.py]: Fixed by cf8589323c0169e654c82fe83ec000adf97f783c

**Pull request merged**